src/Symfony/Component/Cache/Adapter/Php7Adapter.php

```diff
@@ -0,0 +1,376 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "store" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class Php7Adapter implements AdapterInterface
+{
+    private $file;
+    private $umask;
+
+    /**
+     * @var array
+     */
+    private $values;
+
+    /**
+     * @var \Closure
+     */
+    private $createCacheItem;
+
+    /**
+     * @var AdapterInterface
+     */
+    private $fallbackPool;
+
+    /**
+     * @param string $file The PHP file were values are cached.
+     *
+     * @return CacheItemPoolInterface
+     */
+    public static function createOnPhp7($file, CacheItemPoolInterface $fallbackPool)
+    {
+        // OPcache shared memory for arrays works only with PHP 7.0+
+        if (PHP_VERSION_ID >= 70000) {
+            if (!$fallbackPool instanceof AdapterInterface) {
+                $fallbackPool = new ProxyAdapter($fallbackPool);
+            }
+
+            return new static($file, $fallbackPool);
+        }
+
+        return $fallbackPool;
+    }
+
+    /**
+     * @param string           $file         The PHP file were values are cached.
+     * @param AdapterInterface $fallbackPool A pool to fallback on when an item is not hit.
+     * @param int              $umask
```

Comment:
IMHO we should drop this umask concern: there is no other place in the Symfony code base where we deal with this, which means that we've always considered that a wrong system umask (zero), is an ops mistake, and it's not Symfony's job to try to fix external mistakes. We've had the same reasonings about e.g. Varnish misconfigs.